### DISCUSSION TITLE
因在第三方系统里使用，Row和Col之间莫名的隔了两层,且Col没有继承到Row的gutter，这会实际影响布局的表现吗？

### DISCUSSION THREAD

**Discussion by szqyqxgs** [Tencent/tdesign-react#1163] [Q&A] [answered]
![1](https://user-images.githubusercontent.com/108825798/180708847-8382f3e3-e8d4-4a0a-831c-277c5d654698.png)

上图是TD官方布局组件的表现，可见：Col紧跟在Row内，无中间层，且继承了Row的gutter属性

![2](https://user-images.githubusercontent.com/108825798/180709537-f009a07f-1f5c-4e7d-a6a0-ad87903b0b01.png)

上图是本人在第三方系统里使用TD布局组件的表现，可见：Col和Row莫名的隔了 n 和 u 两层，Row有gutter属性

![3](https://user-images.githubusercontent.com/108825798/180709591-417f7e5a-be5d-4ec7-8457-1c6895da5c79.png)

如上图，Col却并没有继承到Row的gutter，这会实际影响布局的表现吗？

**Comment by honkinglin** [comment, marked as answer]
会影响到，能看下源代码是怎么写的么？
看起来像是自己封装了个组件然后包了 Col 组件，目前是根据父子节点关系传递Row 的gutter配置

**Comment by szqyqxgs** [comment]
![333](https://user-images.githubusercontent.com/108825798/180722407-79a5bd60-0228-4e44-987a-9711992e2953.png)

是的，如上图，一个Row组件，一个Col组件。

![555](https://user-images.githubusercontent.com/108825798/180723096-741f69a6-64e5-428b-930e-e12f11bab578.png)

做栅格就是在Row组件下添加Col组件

![222](https://user-images.githubusercontent.com/108825798/180722434-040b6a4d-9ffa-43ba-b76a-88d7cb86b5a8.png)

Row组件的 jsx 部分内容如上图

![111](https://user-images.githubusercontent.com/108825798/180722467-e68359a6-6ce4-470c-82e8-8b2940cd0431.png)

Col 组件的 jsx 部分内容如上图。

那请问我该怎么做呢？
 我需要联系第三方系统让他们解决，还是你们解决？ 还是说这个情况就没法使用TD的栅格了呢？

**Reply by honkinglin** [reply]
其实我也可以改成 context 跨层级传递 gutter，我下版本优化下吧，你那边不用改

**Reply by honkinglin** [reply]
https://github.com/Tencent/tdesign-react/pull/1167
这个pr 会修复你的问题，如果着急的话我可以发个alpha版本给你先用下

**Reply by szqyqxgs** [reply]
好的，辛苦了，感谢你！

另外想问请问下，还有哪个组件会涉及到 父子节点关系传递的? 如果有 是否会考虑改成跨层级传递？ （因为按照目前的情况，如果不能跨层级传递我就不能用，我也好有个准备）

**Reply by szqyqxgs** [reply]
不着急，我会关注这个pr

**Reply by honkinglin** [reply]
> 好的，辛苦了，感谢你！
> 
> 另外想问请问下，还有哪个组件会涉及到 父子节点关系传递的? 如果有 是否会考虑改成跨层级传递？ （因为按照目前的情况，如果不能跨层级传递我就不能用，我也好有个准备）

看了一下，Steps、Swiper、Collapse 跟各自的 Item 还是依赖父子节点关系，不过这三个本身也不建议在 中间掺杂别的dom元素，应该还好
